src/nvim/api/autocmd.c

```diff
@@ -151,7 +151,11 @@ ArrayOf(DictAs(get_autocmds__ret)) nvim_get_autocmds(Dict(get_autocmds) *opts, A
     }
   }
 
-  VALIDATE((!HAS_KEY(opts, get_autocmds, pattern) || !HAS_KEY(opts, get_autocmds, buffer)),
+  bool has_buffer = HAS_KEY(opts, get_autocmds, buffer);
+  bool has_buf = HAS_KEY(opts, get_autocmds, buf);
+  bool has_any_buffer = has_buffer || has_buf;
```

Comment:
This should make the rest of the code much simpler:

```
Object *buf = HAS_KEY(opts, get_autocmds, buf)
  ? opts->buf
  : (HAS_KEY(opts, get_autocmds, buffer) ? opts->buffer : NULL)
bool has_buf = !!buf;
```

then just use `buf` in the rest of the code, instead of `opts->buf`/`opts->buffer`. hopefully that should work :)